### PR TITLE
boards: arm: Add support for STM32H7B3I_DK board and STM32H7B3 SoCs

### PR DESCRIPTION
**Pull request by tmilkovic51**
Used PR #39295 as a reference for adding a new board

Tested both KSCAN and FMC on this board while developing an LTDC driver (will open a PR for that after I tidy it up a little)

### CODE REVIEW

File: boards/arm/stm32h7b3i_dk/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1 TX/RX : PA9/PA10 (ST-Link Virtual Port Com)
+- UART_4 TX/RX : PB9/PB8 (Arduino Serial)
+- LD1 : PG11
+- LD2 : PG2
+- USER_PB : PC13
+
```
Comment (erwango) [inline boards/arm/stm32h7b3i_dk/doc/index.rst:85]: Would be nice to add FMC here. If too much tedious, let's point to board dts file as reference

**Review by erwango — APPROVED** [review]
Thanks for this work.
Minor comment but approved

**Review by FRASTM — DISMISSED** [review]

**Comment by tmilkovic51** [comment]
> Thanks for this work. Minor comment but approved

@erwango  Thank you for your time, I made the changes you suggested.

There is still one open question regarding splitting the STM32H7B3 dtsi files per flash size (like for other H7 SoCs). Currently H7B3 line only comes with 2 MB of flash so it may be unnecessary, but defining the flash size in `stm32h7b3Xi.dtsi` may be more future-proof. Shall I also include that change?

**Comment by erwango** [comment]
> There is still one open question regarding splitting the STM32H7B3 dtsi files per flash size (like for other H7 SoCs). Currently H7B3 line only comes with 2 MB of flash so it may be unnecessary, but defining the flash size in stm32h7b3Xi.dtsi may be more future-proof. Shall I also include that change?

Thanks for raising this point that I had missed. Yes, please add stm32h7b3Xi.dtsi for consistency.

**Comment by tmilkovic51** [comment]
> > There is still one open question regarding splitting the STM32H7B3 dtsi files per flash size (like for other H7 SoCs). Currently H7B3 line only comes with 2 MB of flash so it may be unnecessary, but defining the flash size in stm32h7b3Xi.dtsi may be more future-proof. Shall I also include that change?
> 
> Thanks for raising this point that I had missed. Yes, please add stm32h7b3Xi.dtsi for consistency.

Done. Tested once more with blinky sample, output device tree and linker script look as expected.

**Review by erwango — DISMISSED** [review]

**Review by ABOSTM — DISMISSED** [review]

File: boards/arm/stm32h7b3i_dk/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| FMC       | on-chip    | memc (SDRAM)                        |
++-----------+------------+-------------------------------------+
+
+
+Other hardware features are not yet supported on Zephyr porting.
```
Comment (MaureenHelm) [inline boards/arm/stm32h7b3i_dk/doc/index.rst:67]:
```suggestion
Other hardware features have not been enabled yet for this board.
```
To be consistent with #41679

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/stm32h7b3i_dk/doc/index.rst (hunk 2 above)
Comment (tmilkovic51) [reply boards/arm/stm32h7b3i_dk/doc/index.rst]: Changed

Comment (MaureenHelm) [reply boards/arm/stm32h7b3i_dk/doc/index.rst]: Thanks. Please squash the fix into the original commit that introduced this file.

Comment (tmilkovic51) [reply boards/arm/stm32h7b3i_dk/doc/index.rst]: Squashed and rebased

Thank you for the quick feedback

**Review by MaureenHelm — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm, thanks for adding this board.

**Review by erwango — APPROVED** [review]

**Pull request merged**